jni/bin/setup.dart

```diff
@@ -154,7 +154,23 @@ void main(List<String> arguments) async {
   } else {
     // pass srcDir absolute path because it will be passed to CMake as arg
     // which will be running in different directory
-    await build(options, srcDir.absolute.path, buildDir.path);
+    final now = DateTime.now().microsecondsSinceEpoch;
+    final tempDirUri = Uri.directory(".dart_tool/cmake-build-$now");
```

Comment:
Why not use the actual temp folder from the OS? Is there a need to have a folder in `.dart_tool`?

Also, `DateTime.now().microsecondsSinceEpoch` is not guaranteed to be unique. https://api.dart.dev/stable/2.18.1/dart-io/Directory/createTemp.html is guaranteed to be unique.

Reply:
It's a small change but before doing that, is there any security best practice I should be aware of before using Directory.systemTemp?

Reply:
No, it's just a directory with a unique name somewhere on disk.

I suggest using `createTemp` and using a prefix for what you're doing. If you use `Directory.systemTemp` you have to invent a unique name yourself.

Reply:
Are you suggesting

```
Directory.systemTemp.createTemp("dartjni_native_build")
```

Or 

```
Directory(".dart_tool").createTemp("cmake_native_build")
```

?

Reply:
Ah, I misunderstood. The first one. (Somehow in my brain createTemp was an abstract method that always used systemTemp under the hood.) Thanks for clarifying!